include/copc-lib/las/point.hpp

```diff
@@ -1,597 +1,590 @@
+        else
+            return scan_angle_rank_;
     }
 
-    void ScanAngleRank(const int8_t &scan_angle)
+    void ScanAngleRank(const int8_t &scan_angle_rank)
     {
-        if (point_10_)
-            point_10_->ScanAngleRank(scan_angle);
-        else
+        if (extended_point_type_)
             throw std::runtime_error("Point14 does not have Scan Angle Rank.");
+        else if (scan_angle_rank < -90 || scan_angle_rank > 90)
+            throw std::runtime_error("Scan Angle Rank must be between -90 and 90");
+        else
+            scan_angle_rank_ = scan_angle_rank;
     }
 
-    int16_t ScanAngle() const
+    int16_t ExtendedScanAngle() const
     {
-        if (point_10_)
-            throw std::runtime_error("Point10 does not have Scan Angle.");
+        if (extended_point_type_)
+            return extended_scan_angle_;
         else
-            return point_14_->ScanAngle();
+            throw std::runtime_error("Point10 does not have Scan Angle.");
     }
 
-    void ScanAngle(const int16_t &scan_angle)
+    void ExtendedScanAngle(const int16_t &scan_angle)
     {
-        if (point_10_)
-            throw std::runtime_error("Point10 does not have Scan Angle.");
+        if (extended_point_type_)
+        {
+            if (scan_angle < -30000 || scan_angle > 30000)
+                throw std::runtime_error("Scan Angle must be between -30000 and 30000");
+            else
+                extended_scan_angle_ = scan_angle;
+        }
         else
-            point_14_->ScanAngle(scan_angle);
+            throw std::runtime_error("Point10 does not have Scan Angle.");
+    }
+
+    float ScanAngle() const
+    {
+        return extended_point_type_ ? 0.006f * float(extended_scan_angle_) : float(scan_angle_rank_);
     }
 
-    uint8_t UserData() const { return point_10_ ? point_10_->UserData() : point_14_->UserData(); }
-    void UserData(const uint8_t &user_data)
+    void ScanAngle(const float &scan_angle)
     {
-        if (point_10_)
-            point_10_->UserData(user_data);
+        if (extended_point_type_)
+            extended_scan_angle_ = static_cast<int16_t>(scan_angle / 0.006f);
         else
-            point_14_->UserData(user_data);
+        {
+            if (scan_angle < -90 || scan_angle > 90)
+                throw std::runtime_error("Scan Angle Rank must be between -90 and 90");
+            else
+                scan_angle_rank_ = static_cast<int8_t>(scan_angle);
+        }
     }
 
-    uint16_t PointSourceId() const { return point_10_ ? point_10_->PointSourceId() : point_14_->PointSourceId(); }
-    void PointSourceId(const uint16_t &point_source_id)
+    uint8_t UserData() const { return user_data_; }
+    void UserData(const uint8_t &user_data) { user_data_ = user_data; }
+
+    uint16_t PointSourceId() const { return point_source_id_; }
+    void PointSourceId(const uint16_t &point_source_id) { point_source_id_ = point_source_id; }
+
+    const uint16_t *Rgb() const
```

Comment:
We can change Rgb to RGB everywhere it's capitalized (allowed by style guide for abbreviations)

Reply:
Should we allow the user to change this array? Or do you want to make them use the setter?